gaia-header.js

```diff
@@ -305,27 +666,25 @@ module.exports = Component.register('gaia-header', {
   }
 
   /**
-   * .flush-left
+   * .cant-center
    *
-   * When the fitted text is flush with the
-   * edge of the left edge of the container
-   * we pad it in a bit.
+   * In the rare event that there isn't
+   * enough room to center, it's best we
+   * fallback to left alignment, otherwise
+   * the title just looks broken.
    */
 
-  ::content h1.flush-left {
-    padding-left: 10px;
+  ::content h1.cant-center {
+    text-align: left;
+    padding: 0;
   }
 
-  /**
-   * .flush-right
-   *
-   * When the fitted text is flush with the
-   * edge of the right edge of the container
-   * we pad it in a bit.
-   */
+  ::content:not([action]) h1.cant-center {
+    -moz-padding-start: 10px;
+  }
 
-  ::content h1.flush-right {
-    padding-right: 10px; /* 1 */
+  ::content h1.cant-center:nth-last-child(2) {
```

Comment:
I mean, this is costly if the DOM changes anywhere, because the browser tries to match the CSS. At least here it's scoped to the component so it's not _that_ bad.